godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -348,139 +363,300 @@ macro_rules! impl_float_vector_component_fns {
+                    $(
+                        self.$comp.fposmod(pmod)
+                    ),*
                 )
             }
 
-            pub fn snapped(self, step: Self) -> Self {
+            /// Returns a vector composed of the [`FloatExt::fposmod`] of this vector's components and `modv`'s components.
+            #[inline]
+            pub fn posmodv(self, modv: Self) -> Self {
                 Self::new(
-                    $( self.$comp.snapped(step.$comp) ),*
+                    $(
+                        self.$comp.fposmod(modv.$comp)
+                    ),*
                 )
             }
+
+            /// Returns a new vector with all components rounded to the nearest integer, with halfway cases rounded away from zero.
+            #[inline]
+            pub fn round(self) -> Self {
+                Self::from_glam(self.to_glam().round())
+            }
         }
 
         impl $crate::builtin::math::ApproxEq for $Vector {
+            /// Returns `true` if this vector and `to` are approximately equal.
             #[inline]
+            #[doc(alias = "is_approx_eq")]
             fn approx_eq(&self, other: &Self) -> bool {
                 $( self.$comp.approx_eq(&other.$comp) )&&*
             }
         }
     };
 }
 
-/// Implements common constants and methods for integer type vectors based on their components.
-macro_rules! impl_integer_vector_component_fns {
+/// Implements functions present on 2D vectors.
+macro_rules! impl_vector2x_fns {
     (
         // Name of the vector type.
         $Vector:ty,
         // Type of target component, for example `real`.
-        $Scalar:ty,
-        // Names of the components, with parentheses, for example `(x, y)`.
-        ($($comp:ident),*)
+        $Scalar:ty
     ) => {
         impl $Vector {
-            /// Squared length (squared magnitude) of this vector.
-            ///
-            /// Runs faster than `length`, so prefer it if you need to compare vectors or need the
-            /// squared distance for some formula.
+            /// Returns the aspect ratio of this vector, the ratio of [`Self::x`] to [`Self::y`].
             #[inline]
-            pub fn length_squared(self) -> i64 {
-                let mut val = 0;
-
-                $(
-                    val += self.$comp as i64 * self.$comp as i64;
-                )*
+            pub fn aspect(self) -> real {
+                self.x as real / self.y as real
+            }
 
-                val
+            /// Returns the axis of the vector's highest value. See [`Vector2Axis`] enum. If all components are equal, this method returns [`Vector2Axis::X`].
+            #[inline]
+            #[doc(alias = "max_axis_index")]
+            pub fn max_axis(self) -> $crate::builtin::Vector2Axis {
+                use $crate::builtin::Vector2Axis;
+
+                if self.x < self.y {
+                    Vector2Axis::Y
+                } else {
+                    Vector2Axis::X
+                }
```

Comment:
This changes semantics and signature. Current implementation is:
```rs
    pub fn max_axis(self) -> Option<Vector2Axis> {
        match self.x.cmp(&self.y) {
            Ordering::Less => Some(Vector2Axis::Y),
            Ordering::Equal => None,
            Ordering::Greater => Some(Vector2Axis::X),
        }
    }
```
It's relatively easy to do `.unwrap_or()` on user side, but gives more control about the fallback than the Godot impl.

Handling equality may be especially relevant for the integer vectors.

Reply:
Since `f32` doesn't implement `cmp` should the implementations be split or would this be sufficent:
```rust
pub fn max_axis(self) -> std::option::Option<$crate::builtin::Vector2Axis> {
    use core::cmp::Ordering;
    use $crate::builtin::Vector2Axis;

    match self.x.partial_cmp(&self.y) {
        Some(Ordering::Less) => Some(Vector2Axis::Y),
        Some(Ordering::Equal) => None,
        Some(Ordering::Greater) => Some(Vector2Axis::X),
        _ => None,
    }
}
```

Reply:
I think that should be good enough. The caveat is of course that with float imprecision, exact equality is unlikely as soon as arithmetic is involved, but I think that's fine.

Btw, since the macros are not user-facing but instantiated locally, you don't need to fully qualify all the symbols. Just import them at the beginning of the file -- that makes things quite a bit easier.

What I was wondering also is, should we document how we can get the Godot behavior of `max_axis_index` using this function? With `unwrap_or`?

Reply:
How about?
```rust
/// To mimic Godot's behavior unwrap this function with`unwrap_or(Vector2Axis::Y)`.
```

Reply:
Sounds great, thanks! 

Tiny adjustment (also space/comma):
```
/// To mimic Godot's behavior, unwrap this function's result with `unwrap_or(Vector2Axis::Y)`.
```

Btw, maybe wait with resolving conversations that are still active 🙂